lib/ansible/runner.rb

```diff
@@ -371,7 +371,14 @@ def wait_for(path, timeout: 10.seconds)
           path_created.set if added.include?(path) || modified.include?(path)
         end
 
-        thread = Thread.new { listener.start }
+        thread = Thread.new do
+          listener.start
+        rescue ArgumentError
+          # If the main thread raises an exception immediately it is possible
+          # for the ensure block to call `listener.stop` before this thread
+          # begins its execution resulting in an ArgumentError due to the state
+          # being `:stopped`
```

Comment:
Is it possible to catch this more specifically by looking at something in the ArgumentError itself?

Reply:
you mean like `raise unless err.message.match?(/cannot start from state :stopped/)` ?

Reply:
added :point_down: